### PR TITLE
code monitors: add toggle to include results

### LINKED ISSUE #31265: Code monitoring: Add "Include results" toggle in UI
#31001 added the 'IncludeResults' option and #31212 uses it. We need to add this to the UI when creating/editing code monitor actions.

### PR DESCRIPTION
**Pull request by limitedmage**
⚠️ Blocked on #31580, don't merge this PR until that work item is completed.

Fixes #31265

Recommend reviewing by commit since there was some refactoring needed to consolidated some duplicated code for test messages into `ActionEditor.tsx`.

## Screenshots

image

## Test plan

- Unit tests in Jest for behavior
- Storybook tests for visuals

## App preview:

- [Link](https://sg-web-jp-codemonitorsincluderesults.onrender.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by camdencheek — APPROVED** [review]
🥳

**Review by novoselrok — APPROVED** [review]

File: client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx [hunk 1]
```diff
@@ -18,13 +18,26 @@ export interface ActionEditorProps {
...
+    toggleIncludeResults: (includeResults: boolean) => void
+
     canSubmit?: boolean
     onSubmit: React.FormEventHandler
     onCancel?: React.FormEventHandler
 
     canDelete: boolean
     onDelete: React.FormEventHandler
 
+    // Test action
```
Comment (novoselrok) [inline client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx:31]: Is there a way we can consolidate test action props into fewer props? Added some suggestions below, not blocking.

Suggestions:
- `testCalled` + `testError` + `testLoading` -> `testState: 'called' | 'loading' | Error | undefined`
- `testButtonDisabled` + `testButtonDisabledReason` -> Can we keep just the `testButtonDisabledReason` prop? We should be able to determine if the button is disabled if there is a reason or not.

**Comment by limitedmage** [comment]
@camdencheek  Should we wait for search results in email to be added before checking this in? As this will affect the email action which is enabled for everyone.

File: client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx (hunk 1 above)
Comment (limitedmage) [reply client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx:31]: YES thank you, this was bugging me a lot. I will do this.

**Comment by camdencheek** [comment]
>  Should we wait for search results in email to be added before checking this in?

Oh, yeah. We probably should. Was thinking this was still behind a feature flag

**Comment by limitedmage** [comment]
This PR should be updated once #32227 is checked in so changes to `includeResults` are undone when the cancel button is clicked.

**Review by marisak — APPROVED** [review]
Including search results in code monitor  messages is a great idea! 
- I have a quick suggestion for the toggle label, I think it might be a bit clearer to say `Include search results in message` (or at least briefer). 
- The alternating rows of toggles and buttons are a little clunky; I explored a few ways that this could change ([Figma](https://www.figma.com/file/FlsqgXqE2TPmrGDEw9uFtZ/Code-monitors-improvements?node-id=1890%3A13575)), but I don't feel strongly about any of the solutions.

**Comment by camdencheek** [comment]
@limitedmage Is this ready to go now?

**Comment by limitedmage** [comment]
@camdencheek working on this currently, getting it rebased and ready

**Comment by camdencheek** [comment]
Awesome! Thanks!

**Comment by limitedmage** [comment]
@marisak 

> The alternating rows of toggles and buttons are a little clunky; I explored a few ways that this could change ([Figma](https://www.figma.com/file/FlsqgXqE2TPmrGDEw9uFtZ/Code-monitors-improvements?node-id=1890%3A13575)), but I don't feel strongly about any of the solutions.

I'll leave the design as-is for now. I like the option that keeps the enabled toggle in the title whether the box is open or not, but that requires a bit too much refactoring and I want to get this in before merging becomes even more painful.

**Pull request merged**